exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/transactions/ValidErrorTx.java

```diff
@@ -117,51 +102,45 @@ public int hashCode() {
     return Objects.hash(seed, errorCode, errorDescription);
   }
 
-  static TransactionMessageConverter<ValidErrorTx> converter() {
-    return ValidErrorTx.MessageConverter.INSTANCE;
+  public static TransactionMessageConverter<ValidErrorTx> converter() {
+    return Converter.INSTANCE;
   }
 
-  private enum MessageConverter implements TransactionMessageConverter<ValidErrorTx> {
+  private enum Converter implements TransactionMessageConverter<ValidErrorTx> {
     INSTANCE;
 
     @Override
-    public ValidErrorTx fromMessage(Message message) {
-      checkMessage(message);
-
-      // Unpack the message.
-      ByteBuffer rawBody = message.getBody();
-      try {
-        ValidErrorTxBody body = ValidErrorTxBody.parseFrom(rawBody);
-        long seed = body.getSeed();
-        byte errorCode = (byte) body.getErrorCode();
-        // Convert empty to null because unset error description will be deserialized
-        // as empty string.
-        String errorDescription = Strings.emptyToNull(body.getErrorDescription());
-        return new ValidErrorTx(seed, errorCode, errorDescription);
-      } catch (InvalidProtocolBufferException e) {
-        throw new IllegalArgumentException(e);
-      }
+    public ValidErrorTx fromRawTransaction(RawTransaction rawTransaction) {
+      checkRawTransaction(rawTransaction);
+
+      ValidErrorTxBody body = PROTO_SERIALIZER.fromBytes(rawTransaction.getPayload());
+      long seed = body.getSeed();
+      byte errorCode = (byte) body.getErrorCode();
+      // Convert empty to null because unset error description will be deserialized
+      // as empty string.
+      String errorDescription = Strings.emptyToNull(body.getErrorDescription());
+      return new ValidErrorTx(seed, errorCode, errorDescription);
     }
 
     @Override
-    public BinaryMessage toMessage(ValidErrorTx transaction) {
-      return newQaTransactionBuilder(ID)
-          .setBody(serializeBody(transaction))
-          .buildRaw();
+    public RawTransaction toRawTransaction(ValidErrorTx transaction) {
+      byte[] payload = PROTO_SERIALIZER.toBytes(ValidErrorTxBody.newBuilder()
+          .setSeed(transaction.seed)
+          .setErrorCode(transaction.errorCode)
+          .setErrorDescription(transaction.errorDescription)
```

Comment:
Why is `nullToEmpty` removed? Also, (here and elsewhere) I'd consider keeping `serializeBody` to separate things (but don't insist).